pkg/cmd/codespace/ssh.go

```diff
@@ -449,27 +412,39 @@ func getConfiguredPrivateKeys(
 		return nil, fmt.Errorf("could not load ssh configuration: %w", err)
 	}
 
-	var privateKeyPaths []string
-
 	userHomeDir, _ := os.UserHomeDir()
 
 	configLines := strings.Split(string(configBytes), "\n")
 	for _, line := range configLines {
 		line = strings.TrimSpace(line)
 
 		if strings.HasPrefix(line, "identityfile ") {
-			path := strings.SplitN(line, " ", 2)[1]
+			privateKeyPath := strings.SplitN(line, " ", 2)[1]
 
-			if strings.HasPrefix(path, "~") {
+			if strings.HasPrefix(privateKeyPath, "~") {
 				// os.Stat can't handle ~, so convert it to the real path
-				path = strings.Replace(path, "~", userHomeDir, 1)
+				privateKeyPath = strings.Replace(privateKeyPath, "~", userHomeDir, 1)
 			}
 
-			privateKeyPaths = append(privateKeyPaths, path)
+			// The default configuration includes standard keys like id_rsa or id_ed25519,
+			// but these may not actually exist
+			if _, err := os.Stat(privateKeyPath); err != nil {
+				continue
+			}
+
+			publicKeyPath := privateKeyPath + ".pub"
+			if _, err := os.Stat(publicKeyPath); err != nil {
+				continue
+			}
+
+			return &ssh.KeyPair{
+				PrivateKeyPath: privateKeyPath,
+				PublicKeyPath:  publicKeyPath,
+			}, nil
```

Comment:
nit: can we move this into its own function? it's doing enough and we are already a few levels in on the current function.